firmware/library/L2_HAL/memory/sddriver.hpp

```diff
@@ -0,0 +1,1169 @@
+  //                  and returns the received block
+  //                  This function assumes the block length is locked to
+  //                  512-bytes (as is the case for SDHC/SDXC cards).
+  //                  SD Command 17 and 18 are are used for single and multi
+  //                  block reads, respectively.
+  // @parameter       (uint32_t) address      The address of the starting
+  //                                          block to read from. Note that
+  //                                          this address must be aligned
+  //                                          with a block boundary. By
+  //                                          assuming block size of 512-
+  //                                          bytes, valid addresses must be
+  //                                          multiples of 512.
+  //                                          Some Valid Examples:
+  //                                          0x00000000 --> byte addr 0
+  //                                          0x00000200 --> byte addr 512
+  //                                          0x00000400 --> byte addr 1024
+  //                                          0x00000600 --> byte addr 1536
+  //                                          Some Invalid Examples:
+  //                                          0x00000001 --> byte addr 1
+  //                                          0x000001FF --> byte addr 511
+  //                                          0x00000300 --> byte addr 768
+  // @parameter       (uint8_t*) array        A pointer to an array where
+  //                                          the bytes read will be stored.
+  //                                          The array must be able to hold
+  //                                          a multiple 512 bytes (the SD
+  //                                          standard size of blocks).
+  // @parameter       (~uint16_t) blocks      The number of blocks to read.
+  //                                          Inherently, this number should
+  //                                          be the number of 512-byte
+  //                                          blocks that the above array
+  //                                          can store. If omitted, this
+  //                                          defaults to 1 (i.e. a single
+  //                                          512-byte block).
+  // @returns         (uint8_t) status        Returns the R1-type command
+  //                                          response byte returned from
+  //                                          the card after the initial
+  //                                          read command is sent. See
+  //                                          the official SD documentation
+  //                                          for details on what these
+  //                                          flags mean.
+  // TODO(#160):  Write a function that enables the reading of 1 or more
```

Comment:
Remove all TODOs in this file as they will no longer be relevant after this file is merged.